packages/cubejs-questdb-driver/src/QuestQuery.ts

```diff
@@ -0,0 +1,144 @@
+import { BaseFilter, BaseQuery, ParamAllocator } from '@cubejs-backend/schema-compiler';
+
+const GRANULARITY_TO_INTERVAL: Record<string, string> = {
+  second: 's',
+  minute: 'm',
+  hour: 'h',
+  day: 'd',
+  month: 'M',
+  year: 'Y'
+};
+
+class QuestParamAllocator extends ParamAllocator {
+  public paramPlaceHolder(paramIndex: number) {
+    return `$${paramIndex + 1}`;
+  }
+}
+
+class QuestFilter extends BaseFilter {
+  public orIsNullCheck(column: string, not: string): string {
+    return `${this.shouldAddOrIsNull(not) ? ` OR ${column} = NULL` : ''}`;
+  }
+
+  public setWhere(column: string): string {
+    return `${column} != NULL`;
+  }
+
+  public notSetWhere(column: string): string {
+    return `${column} = NULL`;
+  }
+}
+
+export class QuestQuery extends BaseQuery {
+  public newFilter(filter: any) {
+    return new QuestFilter(this, filter);
+  }
+
+  public newParamAllocator(): ParamAllocator {
+    return new QuestParamAllocator();
+  }
+
+  public concatStringsSql(strings: string[]): string {
+    return `concat(${strings.join(', ')})`;
+  }
+
+  public convertTz(field: string): string {
+    return `to_timezone(${field}, '${this.timezone}')`;
+  }
+
+  public timeStampCast(value: string) {
+    return value;
+  }
+
+  public dateTimeCast(value: string) {
+    return value;
+  }
+
+  public subtractInterval(date: string, interval: string): string {
+    const [number, type] = this.parseInterval(interval);
+    return `dateadd('${type}', ${-number}, ${date})`;
+  }
+
+  public addInterval(date: string, interval: string): string {
+    const [number, type] = this.parseInterval(interval);
+    return `dateadd('${type}', ${number}, ${date})`;
+  }
+
+  public unixTimestampSql(): string {
+    // QuestDB's now() function returns epoch timestamp with microsecond granularity.
+
```

Comment:
Please remove empty line.

Reply:
Addressed in df4d6df2c84c483d0a6aa5d8c51a306b23d6e9ff